TPP.Core/Commands/Definitions/BadgeCommands.cs

```diff
@@ -134,15 +141,150 @@ public async Task<CommandResult> SelectBadge(CommandContext context)
 
         public async Task<CommandResult> Pokedex(CommandContext context)
         {
-            User user = (await context.ParseArgs<Optional<User>>()).OrElse(context.Message.User);
+            (Optional<User> optionalUser, Optional<string> mode_, Optional<User> optionalUser2) =
+                await context.ParseArgs<Optional<User>, Optional<string>,Optional<User>>();
+            User user = optionalUser.IsPresent ? optionalUser.Value : context.Message.User;
             bool isSelf = user == context.Message.User;
-            int numUniqueSpecies = (await _badgeRepo.CountByUserPerSpecies(user.Id)).Count;
-            return new CommandResult
+
+            // Have to additionally check if the string is not empty, otherwise "!dex " would jump right into the else - branch.
+            //   Not sure if this is an actual issue when connected to twitch
+            if( !mode_.IsPresent || string.IsNullOrWhiteSpace( mode_.ToString() ) )
             {
-                Response = isSelf
-                    ? $"You have collected {numUniqueSpecies} distinct Pokémon badge(s)"
-                    : $"{user.Name} has collected {numUniqueSpecies} distinct Pokémon badge(s)"
-            };
+                int numUniqueSpecies = (await _badgeRepo.CountByUserPerSpecies(user.Id)).Count;
+                return new CommandResult
+                {
+                    Response = isSelf
+                        ? $"You have collected {numUniqueSpecies} distinct Pokémon badge(s)"
+                        : $"{user.Name} has collected {numUniqueSpecies} distinct Pokémon badge(s)"
+                };
+            }
+            else
+            {
+                string mode = mode_.ToString();
+                ImmutableSortedDictionary<PkmnSpecies, int> numBadgesPerSpecies =
+                    await _badgeRepo.CountByUserPerSpecies( user.Id );
+
+                if( mode.Equals( PokedexModeMissing ))
+                {
+                    IEnumerable<PkmnSpecies> missingList = PokedexData.Load().KnownSpecies.Except( numBadgesPerSpecies.Keys );
+                    // Seems like PokedexData is not sorted. So we have to sort the list.
+                    IEnumerable<string> badgesFormatted = missingList.OrderBy( entry => entry.Id ).Select( entry => $"{entry}" );
+                    return new CommandResult
+                    {
+                        Response = isSelf
+                            ? $"You are currently missing the following badge(s): {string.Join(", ", badgesFormatted)}"
+                            : $"{user.Name} is currently missing the following badge(s): {string.Join(", ", badgesFormatted)}",
+                        ResponseTarget = ResponseTarget.WhisperIfLong
+                    };
+                }
+                else if( mode.Equals( PokedexModeDupes ) )
+                {
+                    var dupeList = numBadgesPerSpecies.Where(kvp => kvp.Value > 1);
+                    if( !dupeList.Any() )
```

Comment:
This is probably a bug. `dupeList` is an `IEnumerable<string>`. `IEnumerable` can only be enumerated once. `Any()` starts enumerating it, possibly causing the first element to be gone for the `dupeList.Select(...)` later on.

Easy solution for these cases is to turn it into a list first using `.ToList()`.

The same applies to basically any calls to `.Any()` in your changes

Reply:
I fixed the any() stuff by converting everything to lists. I will be more careful with "Enumerable" Lists in the future